### ISSUE TITLE
Insufficient permission for adding an object to repository database ./objects

### ISSUE THREAD

**Issue by fdcastel** [bitnami/bitnami-docker-phabricator#49]
I'm getting the following error whenever I push to any repository

```
remote: error: insufficient permission for adding an object to repository database ./objects
remote: fatal: failed to write object
error: unpack failed: unpack-objects abnormal exit
```

What should be the correct settings for `/bitnami/phabricator/repo` folder ?

Currently, mine is like this:

```
drwxrwxr-x 6 phabricator phabricator 4096 Apr 30  2016 1
drwxrwxr-x 6 phabricator phabricator 4096 Apr 30  2016 2
drwxrwxr-x 6 phabricator phabricator 4096 Apr 30  2016 3
drwxrwxr-x 6 phabricator phabricator 4096 Apr 30  2016 4
drwxrwxr-x 6 phabricator phabricator 4096 May  3  2016 5
drwxrwxr-x 6 phabricator phabricator 4096 May  6  2016 6
```

These are the 6 repositories I migrated from my old bitnami stack (non-docker) following the instructions I compiled in PR #44 .

I explicitly set the owner/group for entire `/bitnami/phabricator` with

```
chown -R phabricator:phabricator /bitnami/phabricator
```

Worse: I created a new repository to test and it was created as:

```
drwxr-xr-x 7 phabricator root        4096 Feb  3 18:25 7
```

(root?)

But even in this one I can push anything, albeit the error message is different:

```
remote: fatal: Unable to create temporary file '/bitnami/phabricator/repo/7/./objects/pack/tmp_pack_XXXXXX': Permissio
denied
error: unpack failed: index-pack abnormal exit
```

**Comment by fdcastel** [comment]
I did try the nuclear way:

```
cd /bitnami/phabricator
chown -R phabricator:phabricator .
chmod -R g+rwX . 
find . -type d -exec chmod 755 {} \;
find . -type f -exec chmod 644 {} \;
```

to no avail. The error message continues. I don't have the slightest. 😢

**Comment by prydonius** [comment]
Thanks @fdcastel, we'll take a look at this.

cc @juan131

**Comment by azurewraith** [comment]
Possibly related?

```Error updating working copy: Command failed with error #1!
COMMAND
git clone --bare -- '********' '/opt/bitnami/phabricator/repo/1'

STDOUT
(empty)

STDERR
Cloning into bare repository '/opt/bitnami/phabricator/repo/1'...
/bitnami/phabricator/repo/1: Permission denied
```

also `git` wasn't included within the image and I didn't see any documentation on how to deal with that... I installed via the helm chart [here](https://github.com/kubernetes/charts/tree/master/stable/phabricator) image: bitnami/phabricator:2017.02-r0.

**Comment by davidbhlm-github** [comment]
@fdcastel, we continue checking the issue. We will let you know when we get any update.

@azurewraith , related to `git`, it is included in `/opt/bitnami/git`. We have just released a new revision which adds it to the PATH.

Thank you both for your interest here.

**Comment by simPod** [comment]
@azurewraith did you solve the issue? Having the same problem here

**Comment by azurewraith** [comment]
@simPod - I haven't resolved the permissions issue yet, but have not really devoted a ton of time towards it yet.

**Comment by andresbono** [comment]
Hi @fdcastel, @simPod, @azurewraith. We have just identified the issue. The problem is related to how the `phabricator` user interacts with the repositories and the accessible binaries in the `PATH` environment variable.

The fix will be implemented in the next revision of the container. Thanks for your interest.

**Comment by andresbono** [comment]
Hi again @fdcastel, @simPod, @azurewraith. We have just updated `phabricator` with a new release that addresses the fix.

Let us know if it works for you.

**Comment by fdcastel** [comment]
@andresbono & team: I can confirm upgrading to `2017.07-r0` fixed this problem. 🙌 

@azurewraith @simPod : Could you please confirm if this works for you so we can close this issue?

Thanks guys! 👍

**Comment by fdcastel** [comment]
BTW: @andresbono could you please elaborate what was done to fix the problem?

I'm asking this because in #44 (which is currently on hold because of this very problem) I needed the following to make the repositories writable again:


> 10. Fix phabricator directory permissions:
> 
>   ```bash
>   $ docker exec phabricator chown -R phabricator:phabricator /bitnami/phabricator
>   ```

(Please note I was using `2016.46-r4` back then -- and without this the repositories didn't work)

Now I'm guessing I should remove this step from the instructions. What do you think? Could you or anybody else from the team confirm it for me?


P.S.: My `/bitnami/phabricator/` directory is like this now:

```
drwxr-xr-x  5 phabricator phabricator 4096 Jan 20 23:33 conf
drwxr-xr-x 72 daemon      daemon      4096 Feb 16 15:57 data
drwxr-xr-x  9 phabricator phabricator 4096 Feb  3 18:25 repo
```

**Comment by simPod** [comment]
I had the problem with different docker image, however chown `2000:2000` helped me. Might not be the case for this image.

**Comment by joseajp** [comment]
> Now I'm guessing I should remove this step from the instructions. What do you think? Could you or anybody else from the team confirm it for me?

@fdcastel the `repo` dir needs to be writable by the `phabricator` user. Since in the Bitnami Phabricator Stack that repo is owned by `daemon` it is necessary to add this step to your migration guide so you don't need to remove it.

**Comment by azurewraith** [comment]
While the upgrade process for some reason was not as simple as changing the image via helm (see [gist](https://gist.github.com/anonymous/6e04de6860dced6ef512ecfe0cd1b68f)) starting fresh with the new image seemed to work (luckily I wasn't too invested in the instance yet)

Regarding the gist, perhaps because of my existing permissions / repo config?  When I attempted to roll back to the previous image I got the same error message and for whatever reason apache didn't seem to start correctly, the container was running though so I'm not sure what went on there.  I'll log the gist as an issue in kubernetes/charts/stable/phabricator and move on to other things.

so TL;DR - helm upgrade didn't work but that's not your problem necessarily... starting fresh fixed both the permissions issue and the default `git` binary path issue.
